src/shared/components/Paginator.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import type { LinkProps } from "react-router-dom";
+import type { ForwardAttributes, Pagination } from "../types";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import "./styles/Paginator.scss";
+
+export const PaginatorItem: React.FC<ForwardAttributes & LinkProps> = ({
+  className,
+  children,
+  ...props
+}) => {
+  return (
+    <li className={`Paginator_item ${className || ""}`}>
+      <Link {...props} className="Paginator_link">
+        {children}
+      </Link>
+    </li>
+  );
+};
+
+class Paginator<T> extends React.Component<
+  React.PropsWithChildren<
+    ForwardAttributes &
+      (
+        | { pagination?: undefined; url?: undefined }
+        | { pagination: Pagination<T>; url: (page: number) => string }
+      )
+  >
+> {
+  render() {
+    const { className, children, pagination, url, ...props } = this.props;
+    if (pagination !== undefined) {
+      return (
+        <ul className={`Paginator ${className || ""}`} {...props}>
+          {pagination.page > 1 && (
+            <PaginatorItem
+              to={url!(pagination.page - 1)}
+              className="Paginator_prev"
+              title="Previous page"
+              aria-label="Previous page"
+            >
+              <FontAwesomeIcon icon="chevron-left" />
+            </PaginatorItem>
+          )}
+          {new Array(pagination.num_pages).fill(0).map((_, page) => {
```

Comment:
This whole map function could be simplified by renaming `page` to `idx` and using `page` = `idx + 1`.

Reply:
Done: 9a1440ba